AWS and infrastructure blog post (#2)
MarkEWaite/jenkins.io: content/blog/2020/06/2020-06-12-infra-and-aws-donation.adoc

---
layout: post

> title: "AWS and Infrastructure Updates - Hello from Infra"

Reviewer: I do not get "Hello from Infra" TBH. Maybe just "Jenkins infrastructure sponsorship by AWS"?

Reviewer 2: Imho it's not only about AWS but more global hightlight on sponsoring and contribution

Reviewer 2: Personally I would just title it "Infrastructure News"

Reviewer: ```suggestion
title: "Jenkins Infrastructure: Stats, Updates, and AWS sponsorship"
```

or so?

Reviewer 3: +1 I like the updated title suggested here.

tags:
- aws
- community
- infrastructure

> - kubernetes

Reviewer: ```suggestion
```

YAGNI? No K8s details in the blog

author: olblak
opengraph:

>   image: /images/post-images/2020-06-12-infrastructure-opengraph.png

Reviewer: I do not mind, but I am nit sure what does this image represent

Author: I believe it represents the country by country web page access statistics that are listed in the first measurement.  Darker shared represent more distinct visits from that country.  Thus, China, India, and the United States are darker.

Author: It needs to be replaced.  I like your idea of creating an OpenGraph image with the sponsor logos.  I'll experiment with it using a Google presentation.

Reviewer 2: > I do not mind, but I am nit sure what does this image represent

This graph is coming from google analytics and represents jenkins.io visitors by country where the darker is 147724

Reviewer 3: Suggest we just add a caption to say what it is rather than redoing a whole new image.

---

image:/images/post-images/2020-06-12-infra-update/infrastructure-opengraph.png[World map,role="center", float="right", height=231]

> During April 2020:

Reviewer: ```suggestion
The Jenkins project relies heavily on its infrastructure.
We use websites like link:/[www.jenkins.io] and link:https://plugins.jenkins.io/[plugins.jenkins.io], ticketing systems like link:https://issues.jenkins.io/[issues.jenkins.io]  , CI/CD infrastructure like link:https://ci.jenkins.io/[ci.jenkins.io] , and many other services.
Just to provide some context about the Jenkins infrastructure scale, here are some stats from April 2020:
```

* Over 600 000 people visited link:https://www.jenkins.io/[www.jenkins.io]
* Over 250 000 Jenkins servers regularly checked the
  link:https://pkg.jenkins.io/[Jenkins package server] and the
  link:https://updates.jenkins.io/[Jenkins update server]
* Over 43 000 continuous integration jobs ran on link:https://ci.jenkins.io/[ci.jenkins.io]
* Over 950 plugins ran their continuous integration pipelines on link:https://ci.jenkins.io/[ci.jenkins.io]

> All those services require infrastructure.
> The Jenkins project relies heavily on its infrastructure.
> We use websites like link:/[www.jenkins.io] and link:https://plugins.jenkins.io/[plugins.jenkins.io], ticketing systems like link:https://issues.jenkins.io/[issues.jenkins.io]  , CI/CD infrastructure like link:https://ci.jenkins.io/[ci.jenkins.io] , and many other services.
> And like any organization, there are specific people who make sure that those services are always up and running.

Reviewer: ```suggestion
```

I suggest to move it up

> The Jenkins project, as an open source project, is built and maintained by its awesome community of users.

Reviewer: ```suggestion
The Jenkins project, as an open source project, is built and maintained by its awesome community.
Like in any organization, there are specific people who make sure that those services are always up and running.
```

> Everyone is welcome to participate.

Reviewer: ```suggestion
Everyone is welcome to link:/participate[participate].
```

> Infrastructure is no exception.

Reviewer: ```suggestion
Infrastructure is no exception, we are always link:/projects/infrastructure/#contributing[looking for new contributors] to the infrastructure!
```

> We have a vision to be a community effort trying to be as transparent as possible.

Reviewer: ```suggestion
While we can’t share publicly everything like secrets, certificates,…​ 
we still try to be as transparent as possible so that everybody can understand and improve our infrastructure without having privileged access.
```

Took from https://www.jenkins.io/projects/infrastructure/#contributing

> What better way than using git to manage infrastructure work?

Reviewer: ```suggestion
What better way than using Git to manage infrastructure work?
```

== Who said GitOps?

Since the creation of the link:https://github.com/jenkins-infra[Jenkins-infra organization] on GitHub in March 2008, more than 650 people have contributed to over 80 git repositories.
Those contributions make the Jenkins community what it is today.
If you can't find something there, it probably means that some help is welcomed.

> More recently, with help from link:https://github.com/halkeye[Gavin Mogan], link:https://github.com/timja[Tim Jacomb], and link:https://github.com/slide[Alex Earl], big achievements have been possible on many fronts like automating jenkins releases, refreshing link:https://plugins.jenkins.io/[plugins.jenkins.io], adding new agents to link:https://ci.jenkins.io/computer/[ci.jenkins.io], and maintaining our kubernetes cluster.

Reviewer: ```suggestion
More recently, with help from link:https://github.com/halkeye[Gavin Mogan], link:https://github.com/timja[Tim Jacomb], and link:https://github.com/slide[Alex Earl], big achievements have been possible on many fronts like automating Jenkins releases, refreshing link:https://plugins.jenkins.io/[plugins.jenkins.io], adding new agents to link:https://ci.jenkins.io/computer/[ci.jenkins.io], and maintaining our Kubernetes cluster.
```

We thank them for their help and for the infrastructure progress they have enabled.

== Infrastructure at Scale

Running infrastructure at the scale the Jenkins project does is expensive and sometimes quite challenging.
We are fortunate enough to be supported by many leading companies that provide us their expertise, their products, and their support.

Recently, Amazon Web Services donated $60 000 to run Jenkins infrastructure on the AWS cloud.
We're so grateful for their donation and for the flexibility it provides.
We're running Linux agents with AMD64 and ARM64 architectures on AWS.
We're using AWS cloud for our Windows agents.
The generous infrastructure donation from Amazon Web Services has increased our continuous integration capacity and broadened our platform coverage.

== Our Sponsors

> Sponsors of the Jenkins project include

Reviewer: Add an image with all logos? 
It might also make sense to sort the sponsors in alphabetical order, and maybe split into two lists (major contributors and minor ones, similar to https://www.jenkins.io/).

Author: Sorted into the order that is used on www.jenkins.io for the major sponsors and alphabetized the other sponsors in a second sentence.  Hyperlinked to all of them.

Reviewer 3: Think that is nice to have but maybe save for another post so we can ship this one sooner?

> Sponsors of the Jenkins project include

Reviewer: ```suggestion
Sponsors of the Jenkins infrastructure include
```
